spec.bs

```diff
@@ -6,57 +6,882 @@ Status: CG-DRAFT
+    1. Return |proxyMap|.
+  </div>
 
-  Promise<any> clear();
+The {{SharedStorage}} Interface {#shared-storage-interface}
+==========================================================
+The {{SharedStorage}} interface is the base for derived interfaces {{WindowSharedStorage}} and {{WorkletSharedStorage}}, which are exposed to the {{Window}} and the {{SharedStorageWorklet}}, respectively.
 
-  Promise<USVString> selectURL(DOMString name,
+Methods that allow the setting and/or deleting of data are exposed to both the {{Window}} and the {{SharedStorageWorklet}} and hence are declared in the base {{SharedStorage}} interface, although their implementations may vary depending on their [=environment=]. This makes it possible to modify the data in Shared Storage from multiple contexts.
+
+Meanwhile, methods for running {{SharedStorageOperation}}s, along with the {{WindowSharedStorage/worklet}} attribute which is used to call {{Worklet/addModule()}}, are declared in {{WindowSharedStorage}} and exposed to the {{Window}} only, as these are the means by which the {{Window}} interacts with the {{SharedStorageWorklet}}.
+
+On the other hand, methods for getting data from the [=shared storage database=] are declared in {{WorkletSharedStorage}} and exposed to the {{SharedStorageWorklet}} only, in order to carefully control the flow of data read from the [=shared storage database|database=].
+
+<xmp class='idl'>
+  [Exposed=(Window,SharedStorageWorklet)]
+  interface SharedStorage {
+    Promise<any> set(DOMString key,
+                     DOMString value,
+                     optional SharedStorageSetMethodOptions options = {});
+    Promise<any> append(DOMString key,
+                        DOMString value);
+    Promise<any> delete(DOMString key);
+    Promise<any> clear();
+  };
+
+  dictionary SharedStorageSetMethodOptions {
+    boolean ignoreIfPresent = false;
+  };
+</xmp>
+
+  ## The {{WindowSharedStorage}} interface ## {#window}
+
+  The {{WindowSharedStorage}} interface is as follows.
+
+<xmp class='idl'>
+  [Exposed=(Window)]
+  interface WindowSharedStorage : SharedStorage {
+    Promise<any> run(DOMString name,
+                     optional SharedStorageRunOperationMethodOptions options = {});
+    Promise<USVString> selectURL(DOMString name,
                                  FrozenArray<SharedStorageUrlWithMetadata> urls,
                                  optional SharedStorageRunOperationMethodOptions options = {});
 
-  Promise<any> run(DOMString name, optional SharedStorageRunOperationMethodOptions options = {});
+    readonly attribute SharedStorageWorklet worklet;
+  };
+
+  dictionary SharedStorageRunOperationMethodOptions {
+    object data;
+  };
+</xmp>
+
+  ### Window Setter/Deleter Methods ### {#window-setter}
+
+  <div algorithm>
+    For the <dfn method for="WindowSharedStorage">set(|key|, |value|, |options|)</dfn> method, run the following steps [=in parallel=]:
```

Comment:
The algorithm for IDL methods are usually: "The ... method steps are:".
Also, there's no need to mention "in parallel" in any of these steps:
- "Enqueue the following steps" should be sufficient.
- Many steps (e.g. the param validation part) can be put before "Enqueue the following steps". Only the DB access that can take a longer time should be put inside the "Enqueue".

Reply:
I think that I've addressed now.